pkg/delete/internal/slo/delete.go

```diff
@@ -122,6 +122,27 @@ func isAPIErrorStatusNotFound(err error) bool {
 	return apiErr.StatusCode == http.StatusNotFound
 }
 
+func DeleteAll(ctx context.Context, c client) error {
+	items, err := c.List(ctx)
+	if err != nil {
+		return err
+	}
+
+	var retErr error
+	for _, i := range items.All() {
+		var e entry
+		if err := json.Unmarshal(i, &e); err != nil {
+			retErr = errors.Join(retErr, err)
+			continue
+		}
+		err := deleteSingle(ctx, c, pointer.DeletePointer{Type: string(config.ServiceLevelObjectiveID), OriginObjectId: e.ID})
+		if err != nil {
+			retErr = errors.Join(retErr, err)
+		}
+	}
+	return retErr
```

Comment:
I think we can build a better error with no extra lines of code:
```suggestion
	var errs []error
	for _, i := range items.All() {
		var e entry
		if err := json.Unmarshal(i, &e); err != nil {
			errs = append(errs, err)
			continue
		}
		err := deleteSingle(ctx, c, pointer.DeletePointer{Type: string(config.ServiceLevelObjectiveID), OriginObjectId: e.ID})
		if err != nil {
			errs = append(errs, err)
		}
	}
	return errors.Join(errs...)
```

Reply:
Could you please explain it more?